### PR TITLE
refac: Remove old props that ensure backwards compatibility

### PR DESCRIPTION
**Pull request by lasrinnil**
## Description
This PR will only be committed after a cleanup of messages older than one month on the environments.

Verified on Dev_002 in this run https://github.com/Energinet-DataHub/dh3-environments/actions/runs/10364773356

Data that would have held the older properties have been marked as dequeued and will be deleted by retention.

## References

https://app.zenhub.com/workspaces/mosaic-60a6105157304f00119be86e/issues/gh/energinet-datahub/team-mosaic/217

### CODE REVIEW

**Review by MadsDue — APPROVED** [review]

**Review by MadsDue — APPROVED** [review]

File: source/OutgoingMessages.Domain/DocumentWriters/NotifyWholesaleServices/NotifyWholesaleServicesCimJsonDocumentWriter.cs [hunk 1]
```diff
@@ -112,14 +100,13 @@ private void WriteSeries(IReadOnlyCollection<string> marketActivityRecords, Utf8
...
                     }
 
-#pragma warning disable CS0618 // Type or member is obsolete
-                    if (series.SettlementMethod is not null || series.SettlementType is not null)
+                    if (series.SettlementMethod is not null)
                     {
                         writer.WriteObject(
                             "marketEvaluationPoint.settlementMethod",
-                            KeyValuePair.Create("value", series.SettlementType?.Code ?? series.SettlementMethod!.Code));
+                            KeyValuePair.Create("value", series.SettlementMethod?.Code ?? series.SettlementMethod!.Code));
```
Comment (ebbeknudsen) [inline source/OutgoingMessages.Domain/DocumentWriters/NotifyWholesaleServices/NotifyWholesaleServicesCimJsonDocumentWriter.cs:107]: I don't think `series.SettlementMethod?.Code ?? series.SettlementMethod!.Code` makes any sense? Unless I'm missing something i think you just meant `series.SettlementMethod!.Code`

File: source/Tests/Domain/OutgoingMessages/Queueing/OutgoingMessageTests.cs [hunk 2]
```diff
@@ -59,15 +59,12 @@ public class OutgoingMessageTests
...
+            new object?[] { DocumentType.NotifyWholesaleServices, "{\"TransactionId\":\"39843d03-d5e3-4695-b3a4-2d05a93373dc\",\"CalculationVersion\":1,\"GridAreaCode\":\"870\",\"ChargeCode\":\"123\",\"IsTax\":false,\"Points\":[{\"Position\":1,\"Quantity\":100,\"Price\":100,\"Amount\":100,\"QuantityQuality\":null}],\"EnergySupplier\":{\"Value\":\"1234567894444\"},\"ChargeOwner\":{\"Value\":\"1234567897777\"},\"Period\":{\"Start\":\"2023-11-01T00:00:00Z\",\"End\":\"2023-12-01T00:00:00Z\"},\"SettlementVersion\":null,\"QuantityMeasureUnit\":{\"Code\":\"KWH\",\"Name\":\"Kwh\"},\"QuantityUnit\":null,\"PriceMeasureUnit\":{\"Code\":\"KWH\",\"Name\":\"Kwh\"},\"Currency\":{\"Code\":\"DKK\",\"Name\":\"DanishCrowns\"},\"ChargeType\":{\"Code\":\"D02\",\"Name\":\"Fee\"},\"Resolution\":{\"Code\":\"P1M\",\"Name\":\"Monthly\"},\"MeteringPointType\":{\"Code\":\"E17\",\"Name\":\"Consumption\"},\"Name\":\"NonProfiled\"}" },
```
Comment (ebbeknudsen) [inline source/Tests/Domain/OutgoingMessages/Queueing/OutgoingMessageTests.cs:63]: Maybe I'm misreading, or my internal json parser isn't working correctly, but what does the last `\"Name\":\"NonProfiled\"` belong to?

File: source/Tests/Domain/OutgoingMessages/Queueing/OutgoingMessageTests.cs [hunk 3]
```diff
@@ -59,15 +59,12 @@ public class OutgoingMessageTests
...
+            new object?[] { DocumentType.NotifyWholesaleServices, "{\"OriginalTransactionIdReference\":\"941c025a-3c98-48e5-985d-d0c843c4dc49\",\"TransactionId\":\"050e2270-3702-4885-b021-15d22d3c0977\",\"CalculationVersion\":1,\"GridAreaCode\":\"870\",\"ChargeCode\":\"123\",\"IsTax\":false,\"Points\":[{\"Position\":1,\"Quantity\":100,\"Price\":100,\"Amount\":100,\"QuantityQuality\":null}],\"EnergySupplier\":{\"Value\":\"1234567894444\"},\"ChargeOwner\":{\"Value\":\"1234567897777\"},\"Period\":{\"Start\":\"2023-11-01T00:00:00Z\",\"End\":\"2023-12-01T00:00:00Z\"},\"SettlementVersion\":null,\"QuantityMeasureUnit\":{\"Code\":\"KWH\",\"Name\":\"Kwh\"},\"QuantityUnit\":null,\"PriceMeasureUnit\":{\"Code\":\"KWH\",\"Name\":\"Kwh\"},\"Currency\":{\"Code\":\"DKK\",\"Name\":\"DanishCrowns\"},\"ChargeType\":{\"Code\":\"D02\",\"Name\":\"Fee\"},\"Resolution\":{\"Code\":\"P1M\",\"Name\":\"Monthly\"},\"MeteringPointType\":{\"Code\":\"E17\",\"Name\":\"Consumption\"},\"Name\":\"NonProfiled\"}" },
```
Comment (ebbeknudsen) [inline source/Tests/Domain/OutgoingMessages/Queueing/OutgoingMessageTests.cs:66]: Maybe I'm misreading, or my internal json parser isn't working correctly, but what does the last `\"Name\":\"NonProfiled\"` belong to?

File: source/OutgoingMessages.Domain/DocumentWriters/NotifyAggregatedMeasureData/NotifyAggregatedMeasureDataEbixDocumentWriter.cs [hunk 4]
```diff
@@ -129,23 +129,22 @@ protected override async Task WriteMarketActivityRecordsAsync(IReadOnlyCollectio
...
                 {
-                    // TODO: This is keep for backward compatibility. Remove this in next pull request
                     // only codes has length 3
                     await WriteCodeWithCodeListReferenceAttributesAsync(
                             "SettlementMethod",
-                            EbixCode.Of(SettlementMethod.FromCode(timeSeries.SettlementType)),
+                            EbixCode.Of(SettlementMethod.FromCode(timeSeries.SettlementMethod)),
                             writer)
                         .ConfigureAwait(false);
                 }
```
[the remark this answers is not in the document]
Comment (lasrinnil) [reply source/OutgoingMessages.Domain/DocumentWriters/NotifyAggregatedMeasureData/NotifyAggregatedMeasureDataEbixDocumentWriter.cs]: https://github.com/Energinet-DataHub/opengeh-edi/pull/1149/commits/26c7285ecbb52befef791aae4d5888bbb0906f74

File: source/Tests/Domain/OutgoingMessages/Queueing/OutgoingMessageTests.cs (hunk 3 above)
Comment (lasrinnil) [reply source/Tests/Domain/OutgoingMessages/Queueing/OutgoingMessageTests.cs:66]: Note to self: Add SettlementMethod to the whoelsale objects

File: source/Tests/Domain/OutgoingMessages/Queueing/OutgoingMessageTests.cs (hunk 2 above)
Comment (lasrinnil) [reply source/Tests/Domain/OutgoingMessages/Queueing/OutgoingMessageTests.cs]: I have added the SettlementMethod property to the object in my latest commit. https://github.com/Energinet-DataHub/opengeh-edi/pull/1149/commits/a8dae7216c5dbabca5c98c9859afb7b744bea869

File: source/Tests/Domain/OutgoingMessages/Queueing/OutgoingMessageTests.cs (hunk 3 above)
Comment (lasrinnil) [reply source/Tests/Domain/OutgoingMessages/Queueing/OutgoingMessageTests.cs]: https://github.com/Energinet-DataHub/opengeh-edi/pull/1149/commits/a8dae7216c5dbabca5c98c9859afb7b744bea869

**Review by MadsDue — APPROVED** [review]

**Pull request merged**